packages/data/src/world/oot/overworld.yml

```
@@ -299,8 +299,8 @@
     ARROWS: "is_adult"
     MAGIC: "true"
   locations:
-    "Hyrule Field Ocarina of Time": "is_child && has_spiritual_stones && soul_npc(SOUL_NPC_ZELDA)"
-    "Hyrule Field Song of Time": "is_child && has_spiritual_stones && soul_npc(SOUL_NPC_ZELDA)"
+    "Hyrule Field Ocarina of Time": "is_child && has_spiritual_stones && soul_npc(SOUL_NPC_ZELDA) && can_swim"
+    "Hyrule Field Song of Time": "is_child && has_spiritual_stones && soul_npc(SOUL_NPC_ZELDA) && can_swim"
```

Like the LH Bottle, this can be gotten with Iron Boots.
```suggestion
    "Hyrule Field Song of Time": "is_child && has_spiritual_stones && soul_npc(SOUL_NPC_ZELDA) && (can_swim || has_iron_boots)"
```